src/main.ts

```diff
@@ -10,25 +10,40 @@ async function run(): Promise<void> {
     const helmfileWorkDirectory = core.getInput('helmfile-workdirectory');
     const helmVersion = core.getInput('helm-version');
     const helmPlugins = core.getInput('helm-plugins');
+    const helmfileAutoInit = core.getInput('helmfile-auto-init');
 
     core.debug(`helmfile-args: ${helmfileArgs}`);
     core.debug(`helmfile-version: ${helmfileVersion}`);
     core.debug(`helmfile-workdirectory: ${helmfileWorkDirectory}`);
     core.debug(`helm-version: ${helmVersion}`);
     core.debug(`helm-plugins: ${helmPlugins}`);
 
-    core.startGroup('Install helm');
-    await Promise.all([installHelm(helmVersion)]);
-    core.endGroup();
-
     core.startGroup('Install helmfile');
     await Promise.all([installHelmfile(helmfileVersion)]);
     core.endGroup();
 
-    if (helmPlugins.length > 0) {
-      core.startGroup('Install helm plugins');
-      await installHelmPlugins(helmPlugins.split(','));
-      core.endGroup();
+    switch (helmfileAutoInit) {
+      case 'false':
+        core.startGroup('Install helm');
+        await Promise.all([installHelm(helmVersion)]);
+        core.endGroup();
+
+        if (helmPlugins.length > 0) {
+          core.startGroup('Install helm plugins');
+          await installHelmPlugins(helmPlugins.split(','));
+          core.endGroup();
+        }
```

Comment:
If the user adds plugins we should always install them (even if we use init) as there could be plugins we don't install on `helmfile init` but are present and used e.g. in hooks.

Reply:
@lucasfcnunes yeah. make sense.